internal/app/machined/pkg/controllers/cluster/local_affiliate_test.go

```diff
@@ -169,7 +169,10 @@ func (suite *LocalAffiliateSuite) TestCPGeneration() {
 		asrt.Equal("Talos ("+version.Tag+")", spec.OperatingSystem)
 		asrt.Equal(cluster.KubeSpanAffiliateSpec{}, spec.KubeSpan)
 		asrt.NotNil(spec.ControlPlane)
-		asrt.Equal(6445, spec.ControlPlane.APIServerPort)
+
+		if spec.ControlPlane != nil {
+			asrt.Equal(6445, spec.ControlPlane.APIServerPort)
+		}
```

Comment:
IMHO this looks weird given the check above. I think we can actually do:

```suggestion
		asrt.Equal(6445, pointer.SafeDeref(spec.ControlPlane).APIServerPort)
```